mailpoet/assets/js/src/newsletters/campaign-stats/newsletter-stats-info.tsx

```diff
@@ -10,9 +10,110 @@ import { chevronDown, Icon } from '@wordpress/icons';
 import { MailPoet } from 'mailpoet';
 import { Heading } from 'common/typography/heading/heading';
 import { Grid } from 'common/grid';
-import { FilterSegmentTag, SegmentTags } from 'common';
+import { confirmAlert, FilterSegmentTag, SegmentTags } from 'common';
 import { NewsletterType } from './newsletter-type';
 
+const redirectToNewsletterHome = () => {
+  window.location.href = '?page=mailpoet-newsletters';
+};
+
+const getEditorLink = (newsletter: NewsletterType) => {
+  let editorHref = `?page=mailpoet-newsletter-editor&id=${newsletter.id}`;
+  if (
+    MailPoet.FeaturesController.isSupported('gutenberg_email_editor') &&
+    newsletter.wp_post_id
+  ) {
+    editorHref = `admin.php?page=mailpoet-email-editor&postId=${newsletter.wp_post_id}`;
+  }
+
+  return editorHref;
+};
```

Comment:
Hmm 🤔, this is a great suggestion.
I made a quick attempt at implementing the suggestion in this PR, but I realized the scope might be larger than I expected. 

We do not return `preview_url` in [in NewsletterResponseBuilder](https://github.com/mailpoet/mailpoet/blob/53bc5a25059b700aca87b951edab83e8f9f06e17/mailpoet/lib/API/JSON/ResponseBuilders/NewslettersResponseBuilder.php#L154) but instead in [Newsletters API endpoint](https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/lib/API/JSON/v1/Newsletters.php#L151). I don't know why yet.

It occurred to me we might need to build a service to handle this and also update this section on [the Premium plugin](https://github.com/mailpoet/mailpoet-premium/blob/trunk/lib/API/JSON/v1/ResponseBuilders/StatsResponseBuilder.php#L32-L47) to reuse the service.

Reply:
@triple0t Thank you for exploring the scope of the change.

> We do not return preview_url in [in NewsletterResponseBuilder](https://github.com/mailpoet/mailpoet/blob/53bc5a25059b700aca87b951edab83e8f9f06e17/mailpoet/lib/API/JSON/ResponseBuilders/NewslettersResponseBuilder.php#L154) but instead in [Newsletters API endpoint](https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/lib/API/JSON/v1/Newsletters.php#L151). I don't know why yet.

Aha! I [saw it in the NewsletterResponseBuilder](https://github.com/mailpoet/mailpoet/blob/553fcfa1b4d0007acd470470301a21bbff333580/mailpoet/lib/API/JSON/ResponseBuilders/NewslettersResponseBuilder.php#L154) but I missed it is added only when generating response for listing. So we have it in multiple places. It could be nice to refactor as well.

> I made a quick attempt at implementing the suggestion in this PR, but I realized the scope might be larger than I expected.

Let's leave it for an extra ticket. I created [a Jira ticket for that](https://mailpoet.atlassian.net/browse/MAILPOET-5776). Please feel free to add more details in the ticket.